api/api.go

```diff
@@ -31,109 +30,50 @@ var (
 type API struct {
 	handler    http.Handler
 	db         *gorm.DB
-	paypal     *paypalsdk.Client
-	config     *conf.Configuration
-	mailer     mailer.Mailer
+	config     *conf.GlobalConfiguration
 	httpClient *http.Client
 	log        *logrus.Entry
-	assets     assetstores.Store
 	version    string
 }
 
-type JWTClaims struct {
-	ID           string                 `json:"id"`
-	Email        string                 `json:"email"`
-	AppMetaData  map[string]interface{} `json:"app_metadata"`
-	UserMetaData map[string]interface{} `json:"user_metadata"`
-	*jwt.StandardClaims
+// ListenAndServe starts the REST API.
+func (a *API) ListenAndServe(hostAndPort string) error {
+	return http.ListenAndServe(hostAndPort, a.handler)
 }
 
-func (a *API) withToken(ctx context.Context, w http.ResponseWriter, r *http.Request) context.Context {
-	log := getLogger(ctx)
-	config := getConfig(ctx)
-	authHeader := r.Header.Get("Authorization")
-	if authHeader == "" {
-		log.Info("Making unauthenticated request")
-		return ctx
-	}
+// NewAPI instantiates a new REST API using the default version.
+func NewAPI(globalConfig *conf.GlobalConfiguration, config *conf.Configuration, db *gorm.DB) *API {
+	return NewSingleTenantAPIWithVersion(globalConfig, config, db, defaultVersion)
+}
 
-	matches := bearerRegexp.FindStringSubmatch(authHeader)
-	if len(matches) != 2 {
-		log.Info("Invalid auth header format: " + authHeader)
-		unauthorizedError(w, "Bad authentication header")
-		return nil
-	}
+// NewMultiTenantAPI creates a new REST API to serve multiple tenants.
+func NewMultiTenantAPI(globalConfig *conf.GlobalConfiguration, db *gorm.DB) *API {
+	return NewAPIWithVersion(context.Background(), globalConfig, db, defaultVersion)
```

Comment:
This is the multi-tenant case, so loading from the database based on the headers. This is the functionality that is still not completely specified on how it should operate.